nncf/openvino/graph/node_utils.py

```diff
@@ -321,13 +321,19 @@ def get_weight_channel_axes(node: NNCFNode, weights_port_id: int) -> List[int]:
     if node.metatype not in GENERAL_WEIGHT_LAYER_METATYPES:
         raise ValueError("Channel axis cannot be defined for operation without weights.")
 
-    channel_axis = node.metatype.const_channel_axis
+    channel_axes = node.metatype.const_channel_axis
     if node.metatype == OVMatMulMetatype:
         assert isinstance(node.layer_attributes, OVConstantLayerAttributes)
+        assert len(channel_axes) == 1
+        assert channel_axes[0] in [-1, -2]
         const_attrs = node.layer_attributes.const_attrs[weights_port_id]
+        matmul_channel_axis = channel_axes[0]
         if const_attrs["transpose"]:
-            assert len(channel_axis) == 1
-            assert channel_axis[0] in [0, 1]
-            channel_axis = [1 - channel_axis[0]]
-
-    return channel_axis
+            transpose_swap = {-1: -2, -2: -1}
```

Comment:
What about the `transpose_a` & `transpose_b` parameters? Why are the axes hard-coded and based on which conditions?

Reply:
Probably, I don't understand the original question by I will try to answer: 
- `transpose_a` & `transpose_b` parameters are supported for weights. 
- I guided https://docs.openvino.ai/2022.3/openvino_docs_ops_matrix_MatMul_1.html